### DISCUSSION TITLE
http/2 with plain text connection

### DISCUSSION THREAD

**Discussion by belovaf** [ntex-rs/ntex#374] [Q&A] [answered]
Does ntex support http/2 with simple router `App::new().route(...)`?
I saw ntex-h2 examples, but they are not very convenient for real applications.

**Comment by fafhrd91** [comment, marked as answer]
http2 is supported, but only with tls connection and tls acceptor must have alpn configuration

**Reply by belovaf** [reply]
Thank you for so fast response!
Unfortunately we use istio service mesh and we can't use TLS between application and istio-sidecar.
Actix have `bind_auto_h2c` for this functionality.
It would be great if ntext could do this as well.

**Comment by fafhrd91** [comment]
you can use https://docs.rs/ntex/latest/ntex/http/struct.HttpServiceBuilder.html 

ntex::web::server is just wrapper for HttpServerBuilder
